2.md

```diff
@@ -0,0 +1,273 @@
+  to the `import` name of the `from` module's ***Module Exports Namespace***.
+
+- `{ export: string }`
+
+  Links the named `export` property of this module's ***Module Imports
+  Namespace*** directly to its ***Module Exports Namespace***.
+
+- `{ export: string, as: string}`
+
+  Links the named `export` property of this module's ***Module Imports
+  Namespace*** to the named `as` property of this module's ***Module Exports
+  Namespace***.
+
+- `{ export: string, from: string }`
+
+  Links the named `export` property of the `from` module's ***Module Exports
+  Namespace*** to this module's ***Module Exports Namespace***,
+  bypassing this module's ***Module Imports Namespace*** entirely.
+
+- `{ export: string, as: string, from: string }`
+
+  Links the named `export` property of the `from` module's ***Module Exports
+  Namespace*** to the named `as` property of this module's ***Module Exports
+  Namespace***, bypassing this module's ***Module Imports Namespace***
+  entirely.
+
+- `{ importAllFrom: string }`
+
+  Links all of the names exported by the `importAllFrom` module to this
+  module's ***Module Imports Namespace***.
+
+- `{ exportAllFrom: string }`
+
+  Links all of the names exported by the `exportAllFrom` module to this
+  module's ***Module Exports Namespace***, bypassing this module's ***Module
+  Imports Namespace*** entirely.
+
+- `{ exportAllFrom: string, as: string }`
+
+  Links the ***Module Exports Namespace*** of the module `exportAllFrom`
+  to a property named `as` on this module's ***Module Exports Namespace***,
+  bypassing this module's ***Module Imports Namespace*** entirely.
+
+In the absence of a `bindings` property, module machinery presumes an empty
+bindings array.
+
+> Modules without bindings may still have side-effects in global scope.
+
+Dynamic import of a `Module` with a virtual module source induces the memoized
+`importHook` of the `Module` for each binding that has an `from` property.
+When a `Module` instance exists for every transitive dependency, dynamic import
+advances the linkage to all new namespaces, including all links between
+***module [exports] namespaces*** and ***module import namespaces*** according
+to each source's bindings.
+
+In the absence of an `execute` property, dynamic import assumes an empty
+execute function.
+
+> Modules without execution behavior may still have useful export bindings from
+> other modules.
+
+Dynamic import will then execute the working set of modules according
+to the existing rules of ordering, and will call `execute` for each virtual
+module source, providing the linked ***Module Imports Namespace***,
+a dynamic import function bound to the module instance only if the virtual
+module source has a truthy `needsImport` property, and an `import.meta` object
+only if the virtual module source has a `needsImportMeta` property.
+
+## Design questions
+
+Should virtual module sources support emulated JavaScript?
```

Comment:
I believe the answer is yes. This allows bundlers to down-level compiling module blocks to be run in the older browsers although they cannot be used to transfer across Workers.

Please add a link to the discussion https://github.com/tc39/proposal-compartments/issues/70